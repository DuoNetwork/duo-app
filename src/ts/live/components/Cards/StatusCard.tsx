import { Table } from 'antd';
import * as React from 'react';
import * as CST from '../../common/constants';
import { INodeStatus, IPriceStatus, IStatus } from '../../common/types';
import util from '../../common/util';
import { SCard, SCardTitle, STableWrapper } from './_styled';

const { Column } = Table;

interface IProps {
	status: IStatus[];
}

export default class StatusCard extends React.PureComponent<IProps> {
	public render() {
		const { status } = this.props;
		return (
			<SCard
				title={
					<SCardTitle>{(CST.TH_PROCESS + ' ' + CST.TH_STATUS).toUpperCase()}</SCardTitle>
				}
				width="100%"
				margin="0 0px 0 0"
				inlinetype="table"
			>
				<STableWrapper>
					<Table
						dataSource={status.map(s => ({
							key: s.process,
							[CST.TH_PROCESS]: s.process,
							[CST.TH_UPDATED]: util.convertUpdateTime(s.timestamp),
							[CST.TH_PRICE]: (s as IPriceStatus).price || '',
							[CST.TH_VOLUME]: (s as IPriceStatus).volume || '',
							[CST.TH_BLOCK]: (s as INodeStatus).block || ''
						}))}
						pagination={false}
					>
						{[
							CST.TH_PROCESS,
							CST.TH_UPDATED,
							CST.TH_PRICE,
							CST.TH_VOLUME,
							CST.TH_BLOCK
						].map(th => <Column title={th} dataIndex={th} key={th} />)}
					</Table>
				</STableWrapper>
			</SCard>
		);
	}
}
