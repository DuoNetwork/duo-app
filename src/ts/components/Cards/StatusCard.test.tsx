import { shallow } from 'enzyme';
import * as React from 'react';
import StatusCard from './StatusCard';

describe('StatusCard Test', () => {
	describe('Test Snapshot', () => {
		const status = [
			{
				process: 'test',
				timestamp: 123
			}
		];
		it('Test Snapshot', () => {
			const wrapper = shallow(<StatusCard status={status} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
