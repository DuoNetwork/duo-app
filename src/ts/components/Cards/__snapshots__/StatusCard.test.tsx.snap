// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard Test Test Snapshot Test Snapshot 1`] = `
<Styled(Card)
  inlinetype="table"
  margin="0 0px 0 0"
  title={
    <styled.div>
      PROCESS STATUS
    </styled.div>
  }
  width="100%"
>
  <styled.div>
    <Table
      bordered={false}
      className=""
      dataSource={
        Array [
          Object {
            "Block": "",
            "Price": "",
            "Process": "test",
            "Updated": "Long Time Ago",
            "Volume": "",
            "key": "test",
          },
        ]
      }
      indentSize={20}
      loading={false}
      locale={Object {}}
      pagination={false}
      prefixCls="ant-table"
      rowKey="key"
      showHeader={true}
      size="default"
      useFixedHeader={false}
    >
      <Column
        dataIndex="Process"
        key="Process"
        title="Process"
      />
      <Column
        dataIndex="Updated"
        key="Updated"
        title="Updated"
      />
      <Column
        dataIndex="Price"
        key="Price"
        title="Price"
      />
      <Column
        dataIndex="Volume"
        key="Volume"
        title="Volume"
      />
      <Column
        dataIndex="Block"
        key="Block"
        title="Block"
      />
    </Table>
  </styled.div>
</Styled(Card)>
`;
